### ISSUE TITLE
Export plugin installed name as variable

### ISSUE THREAD

**Issue by augustobmoura** [asdf-vm/asdf#814]
Export the installed name of the plugin for using in scripts. Plugins can have different installed name than it's official, so is safer to get it from a variable than assume a constant.

Maybe `ASDF_PLUGIN_NAME`?

I can work on this PR, it should be trivial, giving it's okay to add it

**Comment by HashNuke** [comment]
Is there any specific use-case for having the plugin name in a variable?

Incase what you are referring to as plugin name is the plugin dir name, then there is `ASDF_INSTALL_PATH` which has the install path of the plugin.

**Comment by augustobmoura** [comment]
Yes, I need to recursively install a different version in https://github.com/asdf-vm/asdf-nodejs/pull/176. When installing a alias version I install the canon version and symlink the alias to it so I need to install other version for the same plugin (here a need the installed plugin name):
`asdf install $PLUGIN_NAME $CANON_VERSION`

then link the `asdf where` of the canon version to the install_path:
`ln -s $(asdf where $PLUGIN_NAME $CANON_VERSION)  $ASDF_INSTALL_PATH`

**Comment by Stratus3D** [comment]
@augustobmoura you shouldn't be calling `asdf` commands from inside a plugin's callback script, however I understand why you are doing it. symlinking one version directory to another is definitely not something that is supported and asdf core could break your code at any time.

I have no objection to adding a `ASDF_PLUGIN_NAME` variable but I'm not sure it's necessary.

**Comment by hyperupcall** [comment]
There is more information [here](https://github.com/asdf-vm/asdf/issues/1281#issuecomment-13805032990 about implementing such boilerplate for plugin authors.

`ASDF_PLUGIN_NAME` is arguably less usefull than the mentioned `ASDF_ARCH` or `ASDF_OS` variables

**Comment by Stratus3D** [comment]
@augustobmoura is the `ASDF_PLUGIN_NAME` env variable still needed?

**Comment by Stratus3D** [comment]
I'm going to close this issue due to lack of a response. I don't think a `ASDF_PLUGIN_NAME` environment variable is needed but I have no objection to adding it if someone wants it. PR welcomed!
